### PR TITLE
Add basic batch processing of notifications with tests.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Adds small utility functions for batch processing changed log models to generate notifications
* Adds tests for the utility functions

## References
Prerequisite work for #8188

## Reviewer guidance
I initially implemented this, assuming that I would immediately have to optimize the batch processing - but then thought it might be better to get it in, do some tests, and then optimize if/where it is needed.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/notifications/api.py [hunk 1]
```diff
@@ -483,3 +485,26 @@ def parse_attemptslog(attemptlog):
 
     if notifications:
         save_notifications(notifications)
+
+
+def batch_process_attemptlogs(attemptlog_ids):
```
Comment (jredrejo) [inline kolibri/core/notifications/api.py:490]: appart from the tests, I don't see any place in the code where these batch_process_xxx functions are  used. I guess there's another PR or issue were this is going to be worked, right?

**Review by jredrejo — APPROVED** [review]
We had already commented most of the code changes, everything looks good to me. I only have the doubt to know what 's the goal of the `batch_process_xxx` functions

File: kolibri/core/notifications/api.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/notifications/api.py:490]: Yes - the issue this is fixing is dependent on the new morango version being merged into Kolibri, as it relies on being able to add a listener for syncing to get these updated ids.

I just wanted to get the code in first, and seems like it was useful. Will continue to chase down the Windows issues.

Comment (jredrejo) [reply kolibri/core/notifications/api.py:490]: Thanks!
then, ready to merge as soon as the Windows™ issues are solved

**Pull request merged**